tools/lib/subcmd/subcmd-util.h

```diff
@@ -49,17 +49,16 @@ static NORETURN inline void die(const char *err, ...)
 
 static inline void *xrealloc(void *ptr, size_t size)
 {
-	void *ret = realloc(ptr, size);
-	if (!ret && !size)
-		ret = realloc(ptr, 1);
-	if (!ret) {
-		ret = realloc(ptr, size);
-		if (!ret && !size)
-			ret = realloc(ptr, 1);
-		if (!ret)
-			die("Out of memory, realloc failed");
-	}
-	return ret;
+	void *ret;
```

Comment:
This change is inscrutable to me. It looks like it introduces indentation errors. Was this intentional?

Reply:
This change was made by the linux kernel in later versions because on newer gcc versions it fails to compile without this change.

Reply:
Let's leave this out. The medium term solution is to use a matching compiler (like in our fedora 35 container). The long term fix is for us to rebase the proj on the latest kernel.